tests/conftest.py

```diff
@@ -13,8 +13,41 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+
+import pytest
+
+import torch
+from datasets import Dataset
+from lerobot.common.datasets.utils import hf_transform_to_torch
+
 from .utils import DEVICE
 
 
 def pytest_collection_finish():
     print(f"\nTesting with {DEVICE=}")
+
+
+@pytest.fixture
+def hf_dataset():
+    dataset = Dataset.from_dict(
+        {
+            "timestamp": [t * 0.1 for t in range(1, 6)],
+            "index": range(5),
+            "episode_index": [0] * 5,
+        },
+    )
+    dataset.set_transform(hf_transform_to_torch)
+    return dataset
+
+
+@pytest.fixture
+def hf_dataset_3_episodes():
+    dataset = Dataset.from_dict(
+        {
+            "timestamp": [torch.tensor(t * 0.1) for t in range(6)],
+            "index": [torch.tensor(idx) for idx in range(6)],
+            "episode_index": [torch.tensor(0)] * 2 + [torch.tensor(1)] + [torch.tensor(2)] * 3,
+        },
+    )
+    dataset.set_transform(hf_transform_to_torch)
+    return dataset
```

Comment:
As you guessed, I think it could be better to revert the refactor with the fixture. I think it interrupts the flow of the test.

For `hf_dataset_3_episodes` I find it quite weird to have to add `torch.tensor`. Did you understand why?

Reply:
ah okay, so I wrote this code before I figured out that `hf_dataset.set_transform` was used, it is a relic of the past that I should have removed! I fixed the `hf_dataset` but missed fixing `hf_dataset_3_episodes` 

reverted to the version with inline data creation

pleasure working on this with you :saluting_face: :slightly_smiling_face: